pkg/ebpf/c/tracer-ebpf.c

```diff
@@ -501,8 +515,39 @@ static void cleanup_tcp_conn(struct pt_regs* ctx, conn_tuple_t* tup) {
         conn.conn_stats = *cst;
     }
 
-    // Send the connection data to the perf buffer
-    bpf_perf_event_output(ctx, &tcp_close_event, cpu, &conn, sizeof(conn));
+    // Batch TCP closed connections before generating a perf event
+    batch_t *batch_ptr = bpf_map_lookup_elem(&tcp_close_batch, &cpu);
+    if (batch_ptr == NULL) {
+        return;
+    }
+
+    // TODO: Can we turn this into a macro based on TCP_CLOSED_BATCH_SIZE?
+    switch (batch_ptr->pos) {
+    case 0:
+        batch_ptr->c0 = conn;
+        batch_ptr->pos++;
+        return;
+    case 1:
+        batch_ptr->c1 = conn;
+        batch_ptr->pos++;
+        return;
+    case 2:
+        batch_ptr->c2 = conn;
+        batch_ptr->pos++;
+        return;
+    case 3:
+        batch_ptr->c3 = conn;
+        batch_ptr->pos++;
+        return;
+    case 4:
+        batch_ptr->c4 = conn;
+        batch_ptr->pos++;
+        return;
+    }
+
+    batch_ptr->c5 = conn;
```

Comment:
I'm a little worried about this logic on low churn systems. There's no upper limit to when we'll flush a particular closed connection as it depends on # of cpus and the connection churn rate.

Can we add a `timestamp` field to the batch structure and also flush if we haven't flushed in 5s?

Reply:
(oh right, this would only get called if a connection is being actively closed... 🤔)

Reply:
Talked in IRL, I'll cut another PR improving the behavior of batching for low churn